application-diagram-api/src/main/java/com/xwiki/diagram/internal/handlers/DiagramContentHandler.java

```diff
@@ -162,48 +156,31 @@ public void updateDiagramContent(XWikiDocument backlinkDoc, DocumentReference or
             linkHandler.updateMxCellNode(mxCellList.item(i), currentDocRef, originalDocRef);
         }
 
-        backlinkDoc.setContent(getStringFromDocument(document));
+        backlinkDoc.setContent(XMLUtils.serialize(document));
         context.getWiki().saveDocument(backlinkDoc, "Updated diagram after page rename", context);
     }
 
-    /**
-     * Convert a document to String.
-     * 
-     * @param doc the Document to be processed
-     * @return converted Document to String
-     */
-    public String getStringFromDocument(Document doc)
-    {
-        try {
-            StringWriter sw = new StringWriter();
-            TransformerFactory.newInstance().newTransformer().transform(new DOMSource(doc), new StreamResult(sw));
-
-            return sw.toString();
-        } catch (Exception ex) {
-            throw new RuntimeException("Error converting to String", ex);
-        }
-    }
-
     /**
      * Search referenced pages inside the content of a diagram.
      * 
      * @param content the content of the diagram
-     * @return includedPages list of pages inside this content
+     * @return linkedPages list of pages linked in this content
      */
-    public List<String> getIncludedPages(String content)
+    public List<DocumentReference> getLinkedPages(String content)
     {
+        List<DocumentReference> linkedPages = new ArrayList<DocumentReference>();
+
         try {
-            GetDiagramLinksHandler getDiagramLinksHandler = new GetDiagramLinksHandler();
             SAXParserFactory.newInstance().newSAXParser().parse(new ByteArrayInputStream(content.getBytes()),
                 getDiagramLinksHandler);
 
-            return getDiagramLinksHandler.getIncludedPages();
+            return getDiagramLinksHandler.getLinkedPages();
         } catch (ParserConfigurationException | SAXException e) {
             e.printStackTrace();
         } catch (IOException e) {
```

Comment:
Multiple ``catch`` blocks make sense if you do something different in each. Otherwise one ``catch`` is enough. And using a logger is much nicer than printing the stack trace like this.